### ISSUE TITLE
[Core feature] Parallelize Node Executions in FlytePropeller

### ISSUE THREAD

**Issue by hamersaw** [flyteorg/flyte#3866]
### Motivation: Why do you think this is important?

Currently, FlytePropeller uses a modified DFS to traverse the FlyteWorkflow DAG where each node evaluation is synchronous. This means that in certain scenarios DAG traversal can be a bottleneck. For example, extremely large workflows or tasks that operate synchronously.

### Goal: What should the final outcome look like, ideally?

FlytePropeller should evaluate nodes in parallel. This could be done adding a layer of futures on top of the current traversal algorithm so that node scheduling remains consistent but the actual evaluation is done in the background.

### Describe alternatives you've considered 

Leaving it as is may result in poor performance in certain scenarios as described.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by hamersaw** [comment]
Closing in favor of a pluggable scheduler in flytepropeller, which will automatically require parallelized node executions.
